### PR TITLE
[Localization] modifier-select-ui-handler Locales

### PR DESCRIPTION
**Pull request by Adri1**
Update of PR #1656 that has never been considered when it was still relevant

## What are the changes?
Locales for item selection menu (modifier-select-ui-handler) between battles

## Why am I doing these changes?
To localize this menu and for accessibility

## What did change?
If a translation is provided, this menu will be in the set language

**Only one minor visual issue :**
The cursor position doesn't adjusts according to text length, it disappears under the text if it is longer than the original text, so if anybody has a solution to provide...
image
image

**[UPDATE]**
Problem solved, thanks to @mercurius-00 
345259531-d1057f7e-be74-4963-b8cf-b8a109291481
345259585-5a481c62-f0fa-4fe5-9436-246f6ed762b0
345259606-619f0e12-a9bf-4021-9bc5-dbea11c3c502

### Screenshots/Videos
**Before** (French)
image

**After** (French)
image
image
image

## How to test the changes?
Being in Classic or Endless with a Lock Capsule and at least one item held by any Pokémon of your team to unlock "Lock rarities" and "Transfer" options in the menu.

## Checklist
- [X] There is no overlap with another PR?
- [X] The PR is self-contained and cannot be split into smaller PRs?
- [X] Have I provided a clear explanation of the changes?
- [X] Have I tested the changes (manually)?
    - [X] Are all unit tests still passing? (`npm run test`)
- [X] Are the changes visual?
  - [X] Have I provided screenshots/videos of the changes?

### CODE REVIEW

File: src/locales/ko/modifier-select-ui-handler.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+export const modifierSelectUiHandler: SimpleTranslationEntries = {
+  "transfer": "Transfer",
+  "reroll": "Reroll",
+  "lockRarities": "Lock Rarities",
+  "checkTeam": "Check Team",
+  "transferDesc": "Transfer a held item from one Pokémon to another.",
+  "rerollDesc": "Spend money to reroll your item options.",
+  "lockRaritiesDesc": "Lock item rarities on reroll (affects reroll cost).",
+  "checkTeamDesc": "Check your team or use a form changing item.",
```
Comment (ddullgi) [inline src/locales/ko/modifier-select-ui-handler.ts:11]:
```suggestion
  "transfer": "건네주기",
  "reroll": "갱신",
  "lockRarities": "희귀도 고정",
  "checkTeam": "파티 확인",
  "transferDesc": "지닌 물건을 다른 포켓몬에게 건네줍니다.",
  "rerollDesc": "돈을 소모하여 아이템 목록을 새로 고칩니다.",
  "lockRaritiesDesc": "갱신되는 아이템의 희귀도가 고정됩니다(갱신 비용 증가).",
  "checkTeamDesc": "파티를 확인하거나 폼 변경 아이템을 사용합니다.",
```

**Review by ddullgi — CHANGES_REQUESTED** [review]

File: src/locales/ko/modifier-select-ui-handler.ts (hunk 1 above)
Comment (ddullgi) [reply src/locales/ko/modifier-select-ui-handler.ts]: It is based on the description of the existing items.

https://github.com/pagefaultgames/pokerogue/blob/db06c53693c655fae1612883cb81397ee00ce95d/src/locales/ko/modifier-type.ts#L195

Reroll -> 갱신 
Rarities -> 희귀도

https://github.com/pagefaultgames/pokerogue/blob/db06c53693c655fae1612883cb81397ee00ce95d/src/locales/ko/tutorial.ts#L15-L18

Team -> 파티

File: src/locales/zh_CN/modifier-select-ui-handler.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+export const modifierSelectUiHandler: SimpleTranslationEntries = {
+  "transfer": "Transfer",
+  "reroll": "Reroll",
+  "lockRarities": "Lock Rarities",
+  "checkTeam": "Check Team",
+  "transferDesc": "Transfer a held item from one Pokémon to another.",
+  "rerollDesc": "Spend money to reroll your item options.",
+  "lockRaritiesDesc": "Lock item rarities on reroll (affects reroll cost).",
+  "checkTeamDesc": "Check your team or use a form changing item.",
```
Comment (40chyan) [inline src/locales/zh_CN/modifier-select-ui-handler.ts:11]:
```suggestion
  "transfer": "交换道具",
  "reroll": "刷新商店",
  "lockRarities": "锁定稀有度",
  "checkTeam": "查看队伍",
  "transferDesc": "将宝可梦携带的道具交换给其他宝可梦",
  "rerollDesc": "花钱刷新道具",
  "lockRaritiesDesc": "在刷新时锁定道具稀有度(影响刷新费用)",
  "checkTeamDesc": "检查队伍或使用形态改变道具",
```

File: src/locales/de/modifier-select-ui-handler.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+  "transfer": "Transfer",
+  "reroll": "Reroll",
+  "lockRarities": "Lock Rarities",
+  "checkTeam": "Check Team",
+  "transferDesc": "Transfer a held item from one Pokémon to another.",
+  "rerollDesc": "Spend money to reroll your item options.",
+  "lockRaritiesDesc": "Lock item rarities on reroll (affects reroll cost).",
+  "checkTeamDesc": "Check your team or use a form changing item.",
+  "rerollCost": "₽{{formattedMoney}}",
+  "itemCost": "₽{{formattedMoney}}"
```
Comment (CodeTappert) [inline src/locales/de/modifier-select-ui-handler.ts:13]:
```suggestion
export const modifierSelectUiHandler: SimpleTranslationEntries = {
  "transfer": "Übertragen",
  "reroll": "Neu rollen",
  "lockRarities": "Seltenheit festsetzen",
  "checkTeam": "Team überprüfen",
  "transferDesc": "Übertrage ein gehaltenes Item zu einem anderen Pokémon.",
  "rerollDesc": "Nutze Geld um die Items neu zu rollen.",
  "lockRaritiesDesc": "Setze die Seltenheit der Items fest. (Beeinflusst die Rollkosten).",
  "checkTeamDesc": "Überprüfe dein Team or nutze Formänderungsitems.",
  "rerollCost": "{{formattedMoney}}₽",
  "itemCost": "{{formattedMoney}}₽"

```

File: src/locales/pt_BR/modifier-select-ui-handler.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+  "transfer": "Transfer",
+  "reroll": "Reroll",
+  "lockRarities": "Lock Rarities",
+  "checkTeam": "Check Team",
+  "transferDesc": "Transfer a held item from one Pokémon to another.",
+  "rerollDesc": "Spend money to reroll your item options.",
+  "lockRaritiesDesc": "Lock item rarities on reroll (affects reroll cost).",
+  "checkTeamDesc": "Check your team or use a form changing item.",
+  "rerollCost": "₽{{formattedMoney}}",
+  "itemCost": "₽{{formattedMoney}}"
```
Comment (josericardo-fo) [inline src/locales/pt_BR/modifier-select-ui-handler.ts:13]:
```suggestion
  "transfer": "Alterar",
  "reroll": "Atualizar",
  "lockRarities": "Travar Raridades",
  "checkTeam": "Checar Time",
  "transferDesc": "Transfira um item segurado de um Pokémon para outro.",
  "rerollDesc": "Gaste dinheiro para atualizar as suas opções de itens.",
  "lockRaritiesDesc": "Trava a raridade dos itens na atualização (afeta o custo da atualização).",
  "checkTeamDesc": "Cheque seu time ou use um item de mudança de forma.",
  "rerollCost": "₽{{formattedMoney}}",
  "itemCost": "₽{{formattedMoney}}"
```

File: src/locales/es/modifier-select-ui-handler.ts [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+  "transfer": "Transfer",
+  "reroll": "Reroll",
+  "lockRarities": "Lock Rarities",
+  "checkTeam": "Check Team",
+  "transferDesc": "Transfer a held item from one Pokémon to another.",
+  "rerollDesc": "Spend money to reroll your item options.",
+  "lockRaritiesDesc": "Lock item rarities on reroll (affects reroll cost).",
+  "checkTeamDesc": "Check your team or use a form changing item.",
+  "rerollCost": "₽{{formattedMoney}}",
+  "itemCost": "₽{{formattedMoney}}"
```
Comment (GoldTra) [inline src/locales/es/modifier-select-ui-handler.ts:13]:
```suggestion
  "transfer": "Transferir",
  "reroll": "Actualizar",
  "lockRarities": "Bloquear Rarezas",
  "checkTeam": "Ver Equipo",
  "transferDesc": "Transfiere un objeto que lleva un Pokémon a otro.",
  "rerollDesc": "Gasta dinero para actualizar las opciones de objetos.",
  "lockRaritiesDesc": "Bloquea las rarezas de los objetos al actualizar (afecta el costo de actualización).",
  "checkTeamDesc": "Revisa tu equipo o usa un objeto que cambia de forma.",
  "rerollCost": "{{formattedMoney}} ₽",
  "itemCost": "{{formattedMoney}} ₽"
```

**Review by GoldTra — CHANGES_REQUESTED** [review]
Spanish translations

File: src/locales/es/modifier-select-ui-handler.ts [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+  "transfer": "Transferir",
+  "reroll": "Actualizar",
+  "lockRarities": "Bloquear Rarezas",
+  "checkTeam": "Ver Equipo",
+  "transferDesc": "Transfiere un objeto que lleva un Pokémon a otro.",
+  "rerollDesc": "Gasta dinero para actualizar las opciones de objetos.",
+  "lockRaritiesDesc": "Bloquea las rarezas de los objetos al actualizar (afecta el costo de actualización).",
+  "checkTeamDesc": "Revisa tu equipo o usa un objeto que cambia de forma.",
+  "rerollCost": "{{formattedMoney}} ₽",
+  "itemCost": "{{formattedMoney}} ₽"
```
Comment (InnocentGameDev) [inline src/locales/es/modifier-select-ui-handler.ts:13]: Small capitalisation change for consistency with the other occurence of the same word. French did the same.

```suggestion
  "transfer": "Transferir",
  "reroll": "Actualizar",
  "lockRarities": "Bloquear rarezas",
  "checkTeam": "Ver Equipo",
  "transferDesc": "Transfiere un objeto que lleva un Pokémon a otro.",
  "rerollDesc": "Gasta dinero para actualizar las opciones de objetos.",
  "lockRaritiesDesc": "Bloquea las rarezas de los objetos al actualizar (afecta el costo de actualización).",
  "checkTeamDesc": "Revisa tu equipo o usa un objeto que cambia de forma.",
  "rerollCost": "{{formattedMoney}} ₽",
  "itemCost": "{{formattedMoney}} ₽"

```

**Review by InnocentGameDev — APPROVED** [review]
Changes seem perfect except for capitalized "Rarezas" which I don't agree with. Thank you @GoldTra

**Review by Tempo-anon — APPROVED** [review]

**Pull request merged**